tools/importTool/src/main/scala/AkkaFileReaderWithActor.scala

```diff
@@ -0,0 +1,138 @@
+        case Failure(e) => {
+          println("Got a failure while reading the file, ", e.getMessage)
+          system.scheduler.scheduleOnce(7000.milliseconds, self, ActorInput)
+        }})
+    httpResponse.foreach({ response =>
+      println("Get response status=" + response.status)
+      val contentLengthHeader = response.headers.find(_.name == "Content-Range").map(_.value).getOrElse("")
+      val fileSize = contentLengthHeader.split("/")(1).toLong + 1
+      bytesAccumulatorActor ! FileSize(fileSize)
+      response.entity.withSizeLimit(fileSize).dataBytes
+        .via(Framing.delimiter(ByteString(System.lineSeparator()), 100000, true))
+        .map(_.utf8String)
+        .map(getTripleKey)
+        .sliding(2, 1)
+        .splitAfter(pair => (pair.head, pair.last) match {
+        case (TripleWithKey(sub1, triple1), TripleWithKey(sub2, triple2)) => sub1 != sub2
+      })
+        .mapConcat(_.headOption.toList)
+        .fold("" -> List.empty[String]) {
+        case ((_, values), TripleWithKey(subject, triple)) => subject -> (triple :: values)
+      }
+        .map(x => x._2)
+        .mergeSubstreams
+        .grouped(25)
+        .mapAsync(1)(batch=> postWithRetry(batch, format, cluster))
+        .map(x=> bytesAccumulatorActor ! Message(x._2))
+        .runWith(Sink.ignore)
+        .onComplete(
+          {
+            case Success(r) => {
+              println("Process completed successfully")
+              system.terminate()
+            }
+            case Failure(e) => {
+              println("Got a failure during the process, ", e.getMessage)
+              system.scheduler.scheduleOnce(7000.milliseconds, self, ActorInput)
+            }})
+
+    })
+  }
+
+  def postWithRetry(batch: Seq[List[String]], format:String, cluster:String): Future[(HttpResponse, Int)] = {
+    retry(10.seconds, 3){ ingest(batch, format, cluster) }
+  }
+
+  def retry[T](delay: FiniteDuration, retries: Int)(task: => Future[(T, Int)])(implicit ec: ExecutionContext, scheduler: Scheduler): Future[(T, Int)] = {
+    task.recoverWith {
+      case e: Throwable if retries > 0 =>
+        println("Failed to ingest,", e.getMessage)
+        println("Going to retry, retry count=" + retries)
+        akka.pattern.after(delay, scheduler)(retry(delay, retries - 1)(task))
+    }
+
+  }
+
+  def getTripleKey(triple:String): TripleWithKey = {
+    TripleWithKey(triple.split(" ")(0), triple)
+  }
+
+
+  def ingest(batch: Seq[List[String]], format:String, cluster:String): Future[(HttpResponse, Int)] = {
+    val infotonsList = batch.flatten
+    val batchSizeInBytes = infotonsList.mkString.getBytes.length + infotonsList.size
+    val postRequest = HttpRequest(
+      HttpMethods.POST,
+      "http://"+ cluster + "/_in?format=" + format,
+      entity = HttpEntity(`text/plain` withCharset `UTF-8`, infotonsList.mkString.getBytes),
+      protocol = `HTTP/1.0`
```

Comment:
leave the protocol the the default one (version 1.1)

Reply:
done